src/utils/hypercore.ts

```diff
@@ -0,0 +1,159 @@
+import type { Chain, Network, Signer } from '@wormhole-foundation/sdk-connect';
+import { routes } from '@wormhole-foundation/sdk-connect';
+import type {
+  PermitDomain,
+  PermitValue,
+  PermitTypes,
+} from '@mayanfinance/swap-sdk';
+import { getHyperCoreUSDCDepositPermitParams } from '@mayanfinance/swap-sdk';
+import type { Eip6963Wallet } from '@wormhole-labs/wallet-aggregator-evm';
+import type {
+  Quote,
+  TransferParams,
+  ValidationResult,
+} from '../routes/mayan/types';
+import { TransferWallet } from 'utils/wallet';
+import type { WormholeConnectWalletProvider } from 'utils/wallet/types';
+import { isEvmChain } from './evm';
+
+// Note: Hyperliquid bridge = Arbitrum bridge + custom payload for USDC deposit to Hyperliquid
+
+type SignerWithProvider<N extends Network> = Signer<N> & {
+  provider?: () => WormholeConnectWalletProvider;
+};
+
+export function isHyperCoreChain(chain: Chain): boolean {
+  return chain === 'HyperCore';
+}
+
+/**
+ * Enforces HyperCore routing rules: only EVM -> HyperCore flows with USDC
+ * inbound, and disallow HyperCore-originated USDC.
+ */
+export function validateHyperCoreTransfer<N extends Network>(
+  request: routes.RouteTransferRequest<N>,
+  params: TransferParams,
+): ValidationResult | null {
+  const { fromChain, toChain } = request;
+
+  if (isHyperCoreChain(fromChain.chain)) {
+    return {
+      valid: false,
+      params,
+      error: new routes.UnavailableError(
+        new Error('HyperCore cannot be used as a source chain'),
+      ),
+    };
+  }
+
+  if (!isHyperCoreChain(toChain.chain)) {
+    return null;
+  }
+
+  if (toChain.network !== 'Mainnet') {
+    return {
+      valid: false,
+      params,
+      error: new routes.UnavailableError(
+        new Error('HyperCore is only available on Mainnet'),
+      ),
+    };
+  }
+
+  // Temporary restriction to EVM for now; SOL and SUI are coming soon
+  if (!isEvmChain(fromChain.chain)) {
+    return {
+      valid: false,
+      params,
+      error: new routes.UnavailableError(
+        new Error('HyperCore only supports EVM source chains'),
+      ),
+    };
+  }
+
+  return null;
+}
+
+/**
+ * Fetches the HyperCore USDC permit params and asks the signer to sign them
+ * when the quote indicates a permit is required.
+ */
+export async function maybeGetHyperCorePermitSignature<N extends Network>(
+  request: routes.RouteTransferRequest<N>,
+  signer: Signer<N>,
+  quote: Quote,
+  destinationAddress: string,
+): Promise<string | undefined> {
+  // Detect when a quote includes HyperCore metadata requiring an EIP-712 permit.
+  const requiresPermit =
+    isHyperCoreChain(request.toChain.chain) &&
+    quote.details?.hyperCoreParams !== undefined;
+
+  if (!requiresPermit) return undefined;
+
+  const quoteDetails = quote.details;
+
+  if (!quoteDetails) {
+    throw new Error('Missing HyperCore quote details required for permit');
+  }
+
+  const arbitrumRpc = (await request.toChain.getRpc()) ?? null;
```

Comment:
request.toChain is HyperCore and not Arb...so its getting HyperCore rpc that doesn't exist. Can you check what this is ending up with?

in connect I think rpc's come from src/sdklegacy/config

Reply:
For portal it will come from https://github.com/wormholelabs-xyz/rpc-proxy-worker/blob/main/src/index.ts
Will have to update there also.

Maybe better would be to force request the ARB rpc instead getRPC(Arbitrum) 

Something like https://github.com/wormhole-foundation/wormhole-connect/blob/e02307fadf4e4ed2105b5a16cefe6166c6487db4/src/hooks/useGetTokenBalances.ts#L187